### DISCUSSION TITLE
Error in latest update

### DISCUSSION THREAD

**Discussion by czimm74** [aaronwmorris/indi-allsky#849] [General]
Told you I would break it ;)

Jul 22 12:32:59 indi-allsky [ERROR] Image-1/MainThread image.processImage() #311: Frame not found: /tmp/tmpjs0xu2h6.dng
Jul 22 12:32:59 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #322: Exposure received in 0.5497 s (0.5496)
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera._processMetadata() #385: libcamera sensor black level key not found
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera._processMetadata() #353: libcamera sensor AWB key not found
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera._processMetadata() #306: libcamera sensor temperature key not found
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera._processMetadata() #283: Metadata file not found: [Errno 2] No such file or directory: '/tmp/tmprhiheoaj.json'
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera.getCcdExposureStatus() #263: libcamera-still error: b'ERROR: *** no cameras available ***\n'
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera.getCcdExposureStatus() #263: libcamera-still error: b'[0:06:57.580901024] [1830] \x1b[1;31mERROR \x1b[1;37mRPI \x1b[1;34mvc4.cpp:219 \x1b[0mFailed to register camera imx477 10-001a: -22\n'
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera.getCcdExposureStatus() #263: libcamera-still error: b'[0:06:57.580883544] [1830] \x1b[1;31mERROR \x1b[1;37mRPI \x1b[1;34mpipeline_base.cpp:818 \x1b[0mFailed to load a suitable IPA library\n'
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera.getCcdExposureStatus() #263: libcamera-still error: b'[0:06:57.580860323] [1830] \x1b[1;31mERROR \x1b[1;37mIPARPI \x1b[1;34mipa_base.cpp:139 \x1b[0mFailed to load tuning data file /usr/share/libcamera/ipa/raspberrypi/imx477_noir.json\n'
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera.getCcdExposureStatus() #263: libcamera-still error: b"[0:06:57.580796938] [1830] \x1b[1;33m WARN \x1b[1;37mRPiController \x1b[1;34mcontroller.cpp:53 \x1b[0mFailed to open tuning file '/usr/share/libcamera/ipa/raspberrypi/imx477_noir.json'\n"
Jul 22 12:32:59 indi-allsky [ERROR] Capture-1/MainThread libcamera.getCcdExposureStatus() #263: libcamera-still error: b'[0:06:57.537820372] [1826] \x1b[1;32m INFO \x1b[1;37mCamera \x1b[1;34mcamera_manager.cpp:297 \x1b[0mlibcamera v0.0.5+83-bde9b04f\n'
Jul 22 12:32:58 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #399: Total time since last exposure 40.0499 s
Jul 22 12:32:58 indi-allsky [INFO] Capture-1/MainThread libcamera.setCcdExposure() #212: image command: libcamera-still --immediate --nopreview --raw --denoise off --gain 1 --shutter 100 --metadata /tmp/tmprhiheoaj.json --metadata-format json --awb daylight --height 1520 --width 2028 --tuning-file /usr/share/libcamera/ipa/raspberrypi/imx477_noir.json --output /tmp/tmpjs0xu2h6.dng
Jul 22 12:32:58 indi-allsky [INFO] Capture-1/MainThread capture.shoot() #1390: Taking 0.00010001 s exposure (gain 1)
Jul 22 12:32:57 indi-allsky [INFO] MainProcess/MainThread allsky._queueManualTasks() #1111: Checking for manually submitted tasks
Jul 22 12:32:52 indi-allsky [INFO] Capture-1/MainThread indi.getTelescopeRaDec() #840: Telescope Coord: RA 7.46, Dec 41.08
Jul 22 12:32:52 indi-allsky [INFO] Capture-1/MainThread capture.detectMoonMode() #1242: Moon altitude: 22:49:23.3, phase 20.6%
Jul 22 12:32:52 indi-allsky [INFO] Capture-1/MainThread capture.detectNight() #1225: Sun altitude: 67:32:41.4
Jul 22 12:32:52 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #205: Exposure state: READY
Jul 22 12:32:52 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #204: Camera last ready: 32.9s
Jul 22 12:32:44 indi-allsky [INFO] MainProcess/MainThread allsky._queueManualTasks() #1111: Checking for manually submitted tasks
Jul 22 12:32:41 indi-allsky [INFO] Capture-1/MainThread indi.getTelescopeRaDec() #840: Telescope Coord: RA 7.46, Dec 41.08
Jul 22 12:32:41 indi-allsky [INFO] Capture-1/MainThread capture.detectMoonMode() #1242: Moon altitude: 22:47:25.8, phase 20.6%
Jul 22 12:32:41 indi-allsky [INFO] Capture-1/MainThread capture.detectNight() #1225: Sun altitude: 67:31:48.7
Jul 22 12:32:41 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #205: Exposure state: READY
Jul 22 12:32:41 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #204: Camera last ready: 21.8s
Jul 22 12:32:30 indi-allsky [INFO] MainProcess/MainThread allsky._queueManualTasks() #1111: Checking for manually submitted tasks
Jul 22 12:32:30 indi-allsky [INFO] Capture-1/MainThread indi.getTelescopeRaDec() #840: Telescope Coord: RA 7.46, Dec 41.08
Jul 22 12:32:30 indi-allsky [INFO] Capture-1/MainThread capture.detectMoonMode() #1242: Moon altitude: 22:45:28.2, phase 20.6%
Jul 22 12:32:30 indi-allsky [INFO] Capture-1/MainThread capture.detectNight() #1225: Sun altitude: 67:30:55.7
Jul 22 12:32:30 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #205: Exposure state: READY
Jul 22 12:32:30 indi-allsky [INFO] Capture-1/MainThread capture.saferun() #204: Camera last ready: 10.6s

Removing the call to the tuning file resolved this error.  But now I see new errors coming up:

Jul 22 12:49:29 indi-allsky [ERROR] Image-17/MainThread image._calculate_8bit_adu() #2083: Detection mask dimensions do not match image
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread image.processImage() #449: Overriding Blue balance: 1.442338
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread image.processImage() #448: Overriding Red balance: 3.222100
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread image.processImage() #359: Image: 4056 x 3040
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread image.debayer() #2220: Overriding CFA pattern: BGGR
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread sqm.calculate() #42: Weighted SQM average: 126412.73
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread sqm.calculate() #37: Raw SQM average: 2899.38
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread sqm._generateSqmMask() #63: Using central 20% ROI for SQM calculations
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread sqm._generateSqmMask() #48: Generating mask based on SQM_ROI
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread sqm.calculate() #21: Exposure: 0.000100, gain: 1
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread image._calibrate() #2011: Dark not found: ccd1 16bit 0.0001000s gain 1 bin 2 0.00c
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread image._calibrate() #1993: Temperature matched dark not found: 0.00c
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread image._calibrate() #1976: Searching for dark frame: gain 1, exposure >= 0.0, temp >= 0.0c
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread image._calibrate() #1964: Bad Pixel Map not found: ccd1 16bit 0.0001000s gain 1 bin 2 0.00c
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread image._calibrate() #1946: Temperature matched bad pixel map not found: 0.00c
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread image._calibrate() #1929: Searching for bad pixel map: gain 1, exposure >= 0.0, temp >= 0.0c
Jul 22 12:49:29 indi-allsky [WARNING] Image-17/MainThread image._detectBitDepth() #1892: Updated default bit depth: 12
Jul 22 12:49:29 indi-allsky [INFO] Image-17/MainThread image._detectBitDepth() #1864: Image max value: 4095
Jul 22 12:49:28 indi-allsky [INFO] Image-17/MainThread image.add() #1787: Image bits: 16, cfa: BGGR
Jul 22 12:49:28 indi-allsky [INFO] Capture-3/MainThread capture.saferun() #322: Exposure received in 2.3176 s (2.3175)
Jul 22 12:49:26 indi-allsky [INFO] Capture-3/MainThread capture.saferun() #399: Total time since last exposure 40.0210 s
Jul 22 12:49:26 indi-allsky [INFO] Capture-3/MainThread libcamera.setCcdExposure() #212: image command: libcamera-still --immediate --nopreview --raw --denoise off --gain -1 --shutter 100 --metadata /tmp/tmpy4mndkm6.json --metadata-format json --awb daylight --height 1520 --width 2028  --output /tmp/tmpeddyvnn_.dng

**Comment by czimm74** [comment]
Removing detection overlay stopped the second batch of errors, but seems like it is ignoring --height and --width now.  Looking into it more

**Comment by aaronwmorris** [comment]
I think the problem is the libcamera system packages upgraded.  You may need to perform a full system package upgrade and perform a full reboot.

```
sudo apt-get update
sudo apt-get dist-upgrade
```

The location of the tuning files changed with the latest version of libcamera.

Old: /usr/share/libcamera/ipa/raspberrypi/
New: /usr/share/libcamera/ipa/rpi/vc4/

**Comment by czimm74** [comment]
Where would you be without me breaking things for you?  ;)

Up, I had some an update and upgrade in the mean time.  Trying the config now and looks like that is fixed.  Now I just have to see if the --height and --width work now.  Then the overlays.

**Reply by czimm74** [reply]
--height and --width seem to be ignored

**Reply by aaronwmorris** [reply]
I would forego using the --height and --width options and use `--mode 2028:1520`

--height/width takes the original sized image and just scales it down in software.  --mode switches the sensor to bin2 and you get a native resolution.  It might require new darks though.

**Comment by czimm74** [comment]
Also noting it is always saying the system is down, even though it is running.

**Reply by aaronwmorris** [reply]
I have noticed that too.  I think that is just a timing thing.  It might take 5 minutes to recognize it is up.

I will look for the problem and put a fix in soon.

**Reply by aaronwmorris** [reply]
I found the problem.  This happened in a recent code refactor.  The WATCHDOG value is supposed to be set on startup, but I left that function call out (by accident).  The watchdog gets set every 3 minutes, so it will eventually recognize it as running.

**Reply by czimm74** [reply]
Took about 20 minutes and now says running.

**Reply by aaronwmorris** [reply]
Merged #850 to for the running status to update at startup.
